samza-yarn/src/main/java/org/apache/samza/job/yarn/YarnClusterResourceManager.java

```diff
@@ -331,9 +332,24 @@ public void launchStreamProcessor(SamzaResource resource, CommandBuilder builder
   public void stopStreamProcessor(SamzaResource resource) {
     synchronized (lock) {
       Container container = allocatedResources.get(resource);
+      /*
+       * 1. Stop the container through NMClient if the container was instantiated as part of NMClient lifecycle.
+       * 2. Stop the container through AMClient by release the assigned container if the container was from the previous
+       *    attempt and managed by the AM due to AM-HA
+       * 3. Ignore the request if the container associated with the resource isn't present in the bookeeping.
+       */
       if (container != null) {
         log.info("Stopping Container ID: {} on host: {}", resource.getContainerId(), resource.getHost());
         this.nmClientAsync.stopContainerAsync(container.getId(), container.getNodeId());
+      } else {
```

Comment:
should we put this block behind the AM-HA config?

Reply:
The call site for this method in the AM HA flow is behind the config. Hence dropping that here for brevity and readability.